### ISSUE TITLE
Setup block does not register relations..

### ISSUE THREAD

**Issue by kwando** [rom-rb/rom#305]
I'm expecting this to work
```ruby
require 'rom/memory'

puts RUBY_DESCRIPTION
puts ROM::VERSION
rom = ROM.setup(:memory){
  relation(:users)
}

# explodes
rom.relation(:users)
```
but it blows up:
```
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
0.9.0
/Users/kwando/Desktop/rom-failure/vendor/bundle/gems/rom-support-0.1.0/lib/rom/support/registry.rb:30:in `block in []': :users doesn't exist in ROM::RelationRegistry registry (ROM::Registry::ElementNotFoundError)
	from /Users/kwando/Desktop/rom-failure/vendor/bundle/gems/rom-support-0.1.0/lib/rom/support/registry.rb:30:in `fetch'
	from /Users/kwando/Desktop/rom-failure/vendor/bundle/gems/rom-support-0.1.0/lib/rom/support/registry.rb:30:in `[]'
	from /Users/kwando/Desktop/rom-failure/vendor/bundle/gems/rom-0.9.0/lib/rom/container.rb:71:in `relation'
	from test.rb:14:in `<main>'
```

It might just be me not being up to speed with the latest things in ROM.

**Comment by kwando** [comment]
```ruby
env = ROM::Environment.new
env.use(:auto_registration)

rom = env.setup(:memory){
  relation(:users)
}

rom.relation(:users)
```
Turns out I have to use the ```:auto_registration``` plugin. Feels kind of silly to have to use that with the setup DSL though.

**Comment by solnic** [comment]
This is fixed / improved via 176bcff991c9de71250c013dec454d4e737fd250 and 4d1bd2e99fb69b32233f34dd6fcedceb828fa2d7
